File: app/src/main/java/com/mdove/levelgame/main/shop/presenter/contract/BlacksmithArmorContract.java
package com.mdove.levelgame.main.shop.presenter.contract;

import com.mdove.levelgame.base.BasePresenter;
import com.mdove.levelgame.base.BaseView;
import com.mdove.levelgame.main.shop.model.mv.BlacksmithModelVM;

import java.util.List;

/**
 * Created by MDove on 2018/10/30.
 */

public interface BlacksmithArmorContract {
    interface IBlacksmithArmorPresenter extends BasePresenter<IBlacksmithArmorView> {
        void initData();

        void onItemBtnClick(String type, Long id);
    }

    interface IBlacksmithArmorView extends BaseView {
        void showData(List<BlacksmithModelVM> data);
    }
}

File: app/src/main/java/com/mdove/levelgame/main/shop/presenter/contract/BlacksmithArmorContract.kt
package com.mdove.levelgame.main.shop.presenter.contract

import com.mdove.levelgame.base.BasePresenter
import com.mdove.levelgame.base.BaseView
import com.mdove.levelgame.main.shop.model.mv.BlacksmithModelVM

/**
 * @author MDove on 2018/11/1
 *
 */
interface BlacksmithAccessoriesContract {
    interface IBlacksmithAccessoriesPresenter : BasePresenter<IBlacksmithAccessoriesView> {
        fun initData()
        fun onItemBtnClick(type: String, id: Long?)
    }

    interface IBlacksmithAccessoriesView : BaseView {
        fun showData(data: ArrayList<BlacksmithModelVM>)
    }
}